src/System Application/App/AI/src/Azure OpenAI/Chat Completion/AOAIToolsImpl.Codeunit.al

```diff
@@ -16,21 +16,57 @@ codeunit 7778 "AOAI Tools Impl"
         AddToolToPayload: Boolean;
         [NonDebuggable]
         ToolChoice: Text;
+#if not CLEAN25
         [NonDebuggable]
         Tools: List of [JsonObject];
         ToolIdDoesNotExistErr: Label 'Tool id does not exist.';
+#endif
         ToolObjectInvalidErr: Label '%1 object does not contain %2 property.', Comment = '%1 is the object name and %2 is the property that is missing.';
         ToolTypeErr: Label 'Tool type must be of function type.';
+        Functions: array[20] of Interface "AOAI Function";
+        FunctionNames: Dictionary of [Text, Integer];
 
+    procedure AddTool(Tool: Interface "AOAI Function")
+    var
+        Index: Integer;
+    begin
+        Initialize();
+        Index := FunctionNames.Count() + 1;
+
+        if Index > ArrayLen(Functions) then
+            Error('Too many tools added. Maximum number of tools is %1', ArrayLen(Functions));
```

Comment:
Maybe consider switching to list?

And instead of `20`, define a variable `AOAIFunctionLimit := 20;`. So the check would be `If Index > AOAIFunctionLimit`